### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by taehoonlee**

### CODE REVIEW

**Comment by taehoonlee** [comment]
This PR includes:
- excluding unnecessary lines (e.g., it is hard or unnecessary to hit `ImportError`)
- excluding legacy layers (all the four layers in `keras/legacy/layers.py` have annotations of "will be removed after 06(or 08)/2017")

The following is a comparison of the test coverages.

- [Before](https://travis-ci.org/keras-team/keras/builds/333105579): the commit where this PR branched from.
- [After](https://travis-ci.org/keras-team/keras/builds/333114268): this PR.

|   | Before | After |
|:-:|---|---|
| Python 2.7 + TF | 88.61% | 89.15% |
| Python 3.6 + TF | 88.52% | 89.05% |
| Python 2.7 + TH | 86.10% | 86.62% |
| Python 3.6 + TH | 86.06% | 86.59% |
| Python 2.7 + C | 85.94% | 86.50% |
| Python 3.6 + C | 85.90% | 86.47% |

The test coverages have increased by 0.52-0.57%. A test coverage of 90% is coming soon!

File: .coveragerc [hunk 1]
```diff
@@ -1,6 +1,16 @@
 [report]
+# Regexes for lines to exclude from consideration
+exclude_lines =
+    # Don't complain about missing debug-only code:
+    def __repr__
```
Comment (fchollet) [inline .coveragerc]: This could be useful to test.

Comment (taehoonlee) [reply .coveragerc]: @fchollet, Updated.

Comment (taehoonlee) [reply .coveragerc]: I would like to remind you that `.coveragerc` is about only calculating test coverages. The `exclude_lines` is not excluding related tests but excluding related missing line counts for reporting test coverages.

File: .coveragerc [hunk 2]
```diff
@@ -1,6 +1,13 @@
 [report]
+# Regexes for lines to exclude from consideration
+exclude_lines =
+    # Don't complain if tests don't hit defensive assertion code:
+    raise ImportError
+    raise NotImplementedError
+
 fail_under = 84
```
Comment (Dref360) [inline .coveragerc]: I would propose to raise this to 85 then.

Comment (taehoonlee) [reply .coveragerc]: Updated.

**Pull request merged**